### ISSUE TITLE
Sync up of dccmonitor and dccvalidator app

### ISSUE THREAD

**Issue by milen-sage** [Sage-Bionetworks/dccvalidator#524]
@danlu1 as we discussed, short-term, we can start working on resolving the sync up issue between dccvalidator and dccmonitor. It would be great if you could setup some time with @afwillia (as per our email thread) and guide him through

- [x] installation
- [x] config
- [x] apps data flows
- [x] where we are currently seeing sync-ing problems wrt how the apps are configured.

This will resolve apps operation short-term.

@ychae this aligns with the onboarding work we already committed to. Over the next couple of months, we'd like to streamline the backend of the dccvalidator and dccmonitor to avoid the cause of these syncing issues altogether on one hand. And on the other, we'd like to consolidate our codebase so that maintenance of our apps across data ingress is streamlined, while users already operating the workflows supported by dccvalidator and dccmonitor do not experience substantial changes. We can discuss + estimate that in more detail in other github issues and sprint planning.  

@ychae as an aside for some reason, I can't add afwillia to the list of assignees; looks like his username is not w/in this repo scope?

**Comment by ychae** [comment]
@milen-sage it's because he's not part of the Sage Org on GitHub yet. I'll file a ticket with IT to get him added.

**Comment by milen-sage** [comment]
Thanks @ychae ! Figure that might have something do with it :)

**Comment by avanlinden** [comment]
Hey @milen-sage and @ychae, is this in response to this issue with dccmonitor/dccvalidator syncing issues I opened a couple of weeks ago? https://github.com/Sage-Bionetworks/dccmonitor/issues/123

(even keeping bug reporting in sync between the apps is kind of a pain)

**Comment by ychae** [comment]
@avanlinden I'm not sure but I think @danlu1 or @milen-sage would know.

**Comment by ychae** [comment]
@milen-sage Anthony's now part of the Sage GitHub org so you should be able to assign him to tickets.

**Comment by milen-sage** [comment]
@avanlinden yes, this is the issue - thank you fore referencing, and we can keep updating there for dccmonitor specific fixes.

However, we'd also like to tackle this sync problem a bit more generally in the context of how the two apps are 1) currently being deployed and used in concert and 2) be able to compare that to the Data Curator app. That way @afwillia can 3) in the future work on both streamlining the configuration of these two apps to minimize steps where they can get out of sync; and eventually consolidate some of their configuration relative to a common way of managing and administering data models and downstream schemas (e.g. an updated version of the process that the Data Curator app currently uses).

We can use this issue (#524) to keep track of 1) - and that will likely take about a month or two. We will open more issues for 2) and 3) - w/ 3) likely requiring 3-4 months of work. Hopefully, we will be able to make progress on #123 over the next 2-3 weeks, depending on holidays.

**Comment by avanlinden** [comment]
@milen-sage @afwillia Awesome, I am open to anything that reduces the mental load of keeping these two apps functioning and up to date. Tag me if there's anything I can help clarify in the dccvalidator extended universe.

**Comment by milen-sage** [comment]
@avanlinden yes, will do! There are a few items around 3) where your input will be for sure needed. 3) interacts with how data model schemas are developed and maintained; and what are the assumptions that the dccvalidator and the dccmonitor make about schema formats and data curation workflow that data contributors go through in different consortia. Definitely something will be working on closely with you.

**Comment by danlu1** [comment]
@afwillia Xa has been informed to add you to the Jumpcloud and VPN scientists group. So you are able to connect to the Shiny server. 
Here is a list of reference you need to setup and config the Shiny App.
1. I[ntroduction to dccvalidator and dccmonitor](https://sagebionetworks.jira.com/wiki/spaces/SBC/pages/1862139908/Start+up+a+Shiny+app).
2. [Config](https://sage-bionetworks.github.io/dccvalidator/articles/customizing-dccvalidator.html#customizing-the-shiny-application)
3. [Connect to Shiny App server via VS code](https://sagebionetworks.jira.com/wiki/spaces/SBC/pages/2527625314/Getting+started+with+AWS+instance+Docker+and+Visual+Studio+Code#:~:text=Connecting%20ShinyApps%20within,User%20%3Cusername%3E)
4. [Manage the apps](https://sagebionetworks.jira.com/wiki/spaces/SBC/pages/2514812944/Manage+PEC+dccvalidator+dccmonitor+data+dictionary)

Here is the [data flows](https://www.figma.com/file/hlcFyoCrqGSdupuVohIU2g/Schema-Ochestration?node-id=0%3A1) chart that we can discuss after you finishing configuration.

**Comment by afwillia** [comment]
@danlu1 I don't see the sysbio folder on lastpass. Can you or IT give me access so I can use the service account?

**Comment by danlu1** [comment]
@afwillia I still wait on IT's reply on if sharing the lastpass is allowed. You mentioned you went through a similar workflow to get the client_secret for Data Curator app. Have you try it for dccvalidator and dccmonitor.

**Comment by ychae** [comment]
@afwillia and/or @danlu1 Would you be able to give me an estimate for how much work is left to resolve this ticket? tyia!

**Comment by afwillia** [comment]
@ychae I just finally got both apps working and can actually start working on the issue. Unfortunately, I don't have a timeline yet, but I'll keep you posted shortly.

**Comment by ychae** [comment]
Great, thanks for the update @afwillia! Please let me know if there's anything you need from me.

**Comment by milen-sage** [comment]
hi @afwillia I updated the issue description, adding some checkboxes - could you confirm if that reflects accurately where we are?

If there are other issues tracking bugs related to the sync-ing issue that @avanlinden and @danlu1 are working on, could we link here?

**Comment by afwillia** [comment]
@milen-sage that looks right. I'm still a little fuzzy on how dccmonitor is getting the templates for validation. The dccmonitor sync bug is still ongoing https://github.com/Sage-Bionetworks/dccmonitor/issues/123

**Comment by milen-sage** [comment]
@afwillia would be great to figure out the validation template for dccmonitor. Is it based on a JSONSchema?

**Comment by avanlinden** [comment]
@afwillia @milen-sage Right now, dccvalidator and dccmonitor use a set of excel templates stored on Synapse for validation.  The AD portal metadata templates are [here](https://www.synapse.org/#!Synapse:syn18512044). The config file specifies the synIDs of the templates that should be used for validation for each configuration; e.g. : https://github.com/Sage-Bionetworks/dccvalidator/blob/1bd6d558eec544ae9a3dcf8e73f600c01e6e3947/inst/config.yml#L99

Before Nicole left, we developed JSON schemas representing the information in these excel templates that are in [this repo](https://github.com/Sage-Bionetworks/sysbioDCCjsonschemas/tree/master/schema_metadata_templates). She also added the functionality for dccvalidator to validate against those JSON schemas (documented in [this PR](https://github.com/Sage-Bionetworks/dccvalidator/pull/491)), but we have not implemented that for AD or PEC yet and still use the excel templates located by synID. 

Anthony, you should be able to access that template folder on Synapse but if you can't for some reason let me know.

**Comment by milen-sage** [comment]
Based on a discussion with @afwillia sounds like the sync problem has been resolved and we can close this issue (and Sage-Bionetworks/dccmonitor#123). 

Thanks @avanlinden, @danlu1 and @afwillia! 

@afwillia will open another issue to keep tabs on how do we use JSONSchema functionality across the dccvalidator and dccmonitor apps, so that it aligns with our broader data model management practices. We can chat about that during the stakeholder meeting @ychae organized.
